src/main/resources/scripts/cleanup/non_prod_primary_and_replica_data_cleanup.sql

```diff
@@ -0,0 +1,82 @@
+--
+-- RWA-2482 cleanup old tasks from primary task database
+--
+
+CREATE OR REPLACE FUNCTION task_cleanup_primary()
+RETURNS VOID
+LANGUAGE plpgsql
+AS $function$
+DECLARE
+    id text;
+BEGIN
+    -- Loop through the rows of the tasks table
+    FOR id IN select task_id from cft_task_db.tasks WHERE EXTRACT(YEAR FROM created) in (2021) LOOP
+
+        -- delete from sensitive_task_event_logs
+        delete from cft_task_db.sensitive_task_event_logs where task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.sensitive_task_event_logs where task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in sensitive_task_event_logs table', id;
+        END IF;
+
+        -- delete from task_roles
+        delete from cft_task_db.task_roles where task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.task_roles where task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in task_roles table', id;
+        END IF;
+
+        -- delete from tasks
+        DELETE FROM cft_task_db.tasks WHERE task_id = id;
+
+        IF EXISTS (SELECT 1 FROM cft_task_db.tasks WHERE task_id = id) THEN
+            RAISE NOTICE 'Issue with the taskId: % in tasks table', id;
+            RETURN;
+        END IF;
+    END LOOP;
```

Comment:
Might have to test this but usually functions run in one transaction and so it all completes or it all fails.  Since you are looping through many records what if 10 pass and the last one fails?  Will the whole thing rollback?

Reply:
I moved the return to the end of the function. This scenario is avoided.

Reply:
Not sure if we are on the same page.  Can discuss.

Reply:
As discussed I added commit at the end the loop. It will commit all the successful tasks until any exception.